survey/src/main/java/com.quickbirdstudios.surveykit/backend/views/question_parts/LocationPickerPart.kt

```diff
@@ -0,0 +1,268 @@
+                            ),
+                            15f
+                        )
+                    )
+                }
+            }
+
+            this.map.setOnCameraMoveStartedListener { reason ->
+                if (reason == GoogleMap.OnCameraMoveStartedListener.REASON_GESTURE) {
+                    locationMarker.animate().translationY(-10f)
+                        .setInterpolator(BounceInterpolator()).setDuration(300)
+                        .start()
+                }
+            }
+
+            this.map.setOnCameraIdleListener {
+                locationMarker.animate().translationY(10f)
+                    .setInterpolator(BounceInterpolator()).setDuration(300)
+                    .start()
+            }
+        }
+        lifecycle.addObserver(this)
+
+        this.addView(search)
+
+        locationMarker.setImageResource(R.drawable.ic_baseline_location_on)
+        frame.addView(mapView, LayoutParams(LayoutParams.MATCH_PARENT, 800))
+        val locationMarkerParams =
+            FrameLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT)
+        locationMarkerParams.gravity = Gravity.CENTER
+        frame.addView(locationMarker, locationMarkerParams)
+
+        this.addView(frame, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT))
+
+        search.threshold = 4
+        search.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {
+                s?.let {
+                    if (!search.isPerformingCompletion) {
+                        addressProvider.input(s.toString())
+                    }
+                }
+            }
+
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+            }
+        })
+
+        search.setOnItemClickListener { adapter: AdapterView<*>, view: View, pos: Int, id: Long ->
+            val addressSuggestion =
+                adapter.adapter.getItem(pos) as AddressSuggestionProvider.AddressSuggestion
+            selected =
+                Selection(addressSuggestion.location.latitude, addressSuggestion.location.longitude)
+            search.dismissDropDown()
+            search.hideKeyboard()
+        }
+
+        addressProvider.onSuggestionListReady = { suggestions ->
+            val adapter: ArrayAdapter<AddressSuggestionProvider.AddressSuggestion> =
+                ArrayAdapter(
+                    search.context,
+                    android.R.layout.simple_list_item_1,
+                    suggestions
+                )
+            search.setAdapter(adapter)
+            search.showDropDown()
+        }
+    }
+
+    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
+        if (ev.actionMasked == MotionEvent.ACTION_DOWN) {
+            val p = parent
```

Comment:
The variable should have a good name, but in this case you can just use `.let{}`.

```kotlin
parent?.let{ nonNullParent ->
    nonNullParent.requestDisallowInterceptTouchEvent(true)
}
```